gitlab/integration_test.go

```diff
@@ -0,0 +1,701 @@
+/*
+Copyright 2020 The Flux CD contributors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package gitlab
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"io/ioutil"
+	"math/rand"
+	"net/http"
+	"os"
+	"sync"
+	"testing"
+	"time"
+
+	"github.com/gregjones/httpcache"
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+	"github.com/xanzy/go-gitlab"
+
+	"github.com/fluxcd/go-git-providers/gitprovider"
+)
+
+const (
+	ghTokenFile = "/tmp/gitlab-token"
+
+	// Include scheme if custom, e.g.:
+	// gitlabDomain = "https://gitlab.acme.org/"
+	// gitlabDomain = "https://gitlab.dev.wkp.weave.works"
```

Comment:
I might have added this comment here:
> Include scheme if custom

But I don't think this should be the case. We shouldn't need to user to include the scheme when declaring custom domain vs. no-domain if using gitlab.com. wdyt?

Reply:
This was the original attempt, but it was added because of go-gitlab returning the error below when the custom domain was passed.
```
Unexpected error:
      <*url.Error | 0xc0002aa4b0>: {
          Op: "Get",
          URL: "gitlab.dev.wkp.weave.works/api/v4/projects/dinos%2Ftestrepo",
          Err: {
              what: "unsupported protocol scheme",
              str: "",
          },
      }
      Get gitlab.dev.wkp.weave.works/api/v4/projects/dinos%2Ftestrepo: unsupported protocol scheme ""
  occurred
```

Reply:
Good point. I guess it comes down to if we wanna support `http`? Which as it hosted internally sometimes is "yes". But it would be nice to support scheme-less urls for "tidyness".

```
c, err = NewClient(gitlabToken, "", WithDomain("gitlab.works")) // "https://gitlab.works"
c, err = NewClient(gitlabToken, "", WithDomain("https://gitlab.works")) // "https://gitlab.works"
c, err = NewClient(gitlabToken, "", WithDomain("http://gitlab.works")) // "http://gitlab.works"
```

Could be a follow up issue.

Reply:
I agree, it could default to `https` and mention it in the error message, only thinking if it's predictable. Will open an issue :+1: